src/components/MyForums.js

```diff
@@ -165,26 +233,105 @@ const MyForums = () => {
             </Box>
 
             <Box sx={{ position: 'absolute', top: 150, left: '50%', transform: 'translateX(-50%)' }}>
-            {forums.map((event) => (
-                <Card style={{ width: '800px', marginBottom: '20px' }} key={event.id}>
-                    <CardContent>
-                        <Typography variant="h5" component="div">
-                            {event.forumTitle}<br />
-                        </Typography>
-                        <Typography sx={{ mb: 1.5 }} color="text.secondary">
-                            Posted on {new Date(event.dateTime).toLocaleDateString()}<br />
-                        </Typography>
-                        <Typography sx={{ mb: 1.5 }} color="text.secondary">
-                            <br />{event.description}<br />
-                        </Typography>
-                    </CardContent>
-                </Card>
-            ))}
+                {forums.map((event) => (
+                    <Card style={{ width: '800px', marginBottom: '20px' }} key={event.id}>
+                        <CardContent>
+                            <Link to={`/forum/${event.id}`} target="_blank">
+                                <Typography variant="h5" component="div">
+                                    {event.forumTitle}<br />
+                                </Typography>
+                            </Link>
+                            <Typography sx={{ mb: 1.5 }} color="text.secondary">
+                                Posted on {new Date(event.dateTime).toLocaleDateString()}<br />
+                            </Typography>
+                            <Typography sx={{ mb: 1.5 }} color="text.secondary">
+                                <br />{event.description}<br />
+                            </Typography>
+                        </CardContent>
+                        <CardActions>
+                            {<Button onClick={() => handleOpenDialog(event)}>Edit Forum</Button>}
```

Comment:
if a forum is archived, should this button appear?

Reply:
I removed the button in the case of archived forums.